fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -386,6 +386,17 @@ public void setLastRefreshTime(long lastRefreshTime) {
     @SerializedName(value = "maxMVRewriteStaleness")
     private int maxMVRewriteStaleness = 0;
 
+
+    // Materialized view's output columns may be different from defined query's output columns.
+    // Record the indexes based on materialized view's column output.
+    // eg: create materialized view mv as select col1, col2, col3 from tbl
+    //  desc mv             :  col2, col1, col3
+    //  queryOutputIndexes  :  1, 0, 2
+    // which means 0th of query output column is in 1th mv's output columns, and 1th -> 0th, 2th -> 2th.
+    @SerializedName(value = "queryOutputIndexes")
+    protected List<Integer> queryOutputIndexes = Lists.newArrayList();
```

Comment:
https://github.com/StarRocks/starrocks/pull/31702. dose this field cause the similar problem above?

Reply:
no. It should not.